consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent/ConsistentParentBasedProbabilitySamplerProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.sampler.consistent;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.traces.ConfigurableSamplerProvider;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+
+public final class ConsistentParentBasedProbabilitySamplerProvider
+    implements ConfigurableSamplerProvider {
+
+  @Override
+  public Sampler createSampler(ConfigProperties config) {
+    double samplingProbability = config.getDouble("otel.traces.sampler.arg", 1.0d);
+    return ConsistentSampler.parentBased(ConsistentSampler.probabilityBased(samplingProbability));
+  }
+
+  @Override
+  public String getName() {
+    return "consistent_parent_based_probability";
```

Comment:
samplers in java sdk use a different naming scheme https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#sampler

Reply:
Would `consistent_parentbased_probability` work?

Reply:
how about

* `consistent_always_on`
* `consistent_always_off`
* `consistent_probability`
* `consistent_rate_limit`
* `parentbased_consistent_always_on`
* `parentbased_consistent_always_off`
* `parentbased_consistent_probability`
* `parentbased_consistent_rate_limit`

(don't need to add them all in this PR, but trying to understand the naming pattern)

Reply:
Sounds good, updated to `parentbased_consistent_probability`

Reply:
could you also rename the sampler class to match the name

Reply:
updated